org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -299,7 +299,7 @@ protected void addProjectSourceFolders(IClasspathDescriptor classpath, Map<Strin
       String mainResourcesEncoding = null;
       String testResourcesEncoding = null;
 
-      boolean isTestCompilationSkipped = false;
+      boolean isTestCompilationSkipped = true;
```

Comment:
In case there is no execution of the compiler plugin configured the source folders will not be configured.
I have not checked if this is a realistic scenario (probably not) but in case it can happen the variable should probably be initialized with
```suggestion
      boolean isTestCompilationSkipped = !executions.isEmpty();
```

Reply:
Right. Then lets keep this as is.

Reply:
OK, as the test have shown the case of no specified execution is definitively relevant.
I now updated this PR to not skip the folder generation if no corresponding test-compile/resource execution is specified.

In general it would be good to have a test-case for your use case in `org.eclipse.m2e.jdt.tests` in the class `JavaConfigurationTest`, analogues to the existing ones.
Can you contribute one similar to your use case?